test/conformance/walk.go

```diff
@@ -14,415 +14,294 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package main provides a tool that scans kubernetes e2e test source code
-// looking for conformance test declarations, which it emits on stdout.  It
-// also looks for legacy, manually added "[Conformance]" tags and reports an
-// error if it finds any.
-//
-// This approach is not air tight, but it will serve our purpose as a
-// pre-submit check.
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"fmt"
 	"go/ast"
 	"go/parser"
 	"go/token"
+	"gopkg.in/yaml.v2"
+	"io"
+	"log"
 	"os"
-	"path/filepath"
 	"regexp"
+	"sort"
 	"strconv"
 	"strings"
 	"text/template"
+
+	"github.com/onsi/ginkgo/types"
 )
 
 var (
-	baseURL                                           = flag.String("url", "https://github.com/kubernetes/kubernetes/tree/master/", "location of the current source")
-	confDoc                                           = flag.Bool("conformance", false, "write a conformance document")
-	version                                           = flag.String("version", "v1.9", "version of this conformance document")
-	totalConfTests, totalLegacyTests, missingComments int
-
-	// If a test name contains any of these tags, it is ineligble for promotion to conformance
-	regexIneligibleTags = regexp.MustCompile(`\[(Alpha|Feature:[^\]]+|Flaky)\]`)
```

Comment:
I have concerns this means we're no longer preventing invalid tests from being promoted to conformance

Reply:
I don't recall removing this part but wouldn't be opposed since the promotion to conformance is a manual process that requires that the same people who administer this list sign off on the promotion.

The only time I've seen a test desire to be promoted that hit this regexp, the regexp was changed to allow the test, not the other way around. It also would motivate situations where you just remove the offending tag in order to promote to work around the issue. 

It also doesn't catch situations where the test lacks the tag in question, even though it would apply (e.g. are all tests properly marked as alpha/feature/flaky). Again, my largest experience with this was that tests that were disruptive weren't marked as such, so the regexp (which included disruptive at the time) didn't block it though it should have.

If you feel strongly I think this would be a pretty simple thing to add back in.

Reply:
Talked about this is the conformance-wg; I will add back in this behavior.